lib/metadata/index.js

```diff
@@ -434,7 +448,7 @@ Metadata.prototype.getUdt = function (keyspaceName, name, callback) {
 Metadata.prototype._getUdtCb = function (keyspaceName, name, callback) {
   let cache;
   if (this.options.isMetadataSyncEnabled) {
-    const keyspace = this.keyspaces[keyspaceName];
+    const keyspace = this._getKeyspaceFromCache(keyspaceName);
```

Comment:
We can't throw an error on an callback-based method, we should manually check, invoke the callback and exit.

Reply:
D'oh, good point!  Don't know why that didn't occur to me at the time.  Also erroring out the callback should also reject the promise, so still solves that for me.  I'll fix it.